### DISCUSSION TITLE
[nullable-analysis] Make a non-nullable parameter, field, or property do not need to be initialized.

### DISCUSSION THREAD

**Discussion by CodingOctocat** [dotnet/roslyn#68758] [Q&A] [answered]
I have a property which is non-nullable and normally I have to initialize it on exit from the constructor and I want to remove this warning.

Since this property is in the ViewModel, it's available through Messenger, so it can't actually be uninitialized when it's actually called.

```CS
public class ViewModel : IRecipient<PropertyChangedMessage<Foo?>>
{
    public Foo MyFoo { get; set; }    

    public void Receive(PropertyChangedMessage<Foo?> message)
    {
        if (message.NewValue is null)
        {
            MyFoo = new Foo();
        }
        else
        {
            MyFoo = message.NewValue;
        }
    }
}
```

How can this warning be removed without introducing new attribute?

**Comment by svick** [comment, marked as answer]
Probably the easiest approach is to assign a default value of `null!` to the property:

```c#
public Foo MyFoo { get; set; } = null!;
```

**Comment by KennethHoff** [comment]
You should also take a look [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis). Specifically [[MemberNotNull]](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.membernotnullattribute?view=net-7.0).

I realize after typing this that this is not really applicable, but it's good to know regardless!
